addons/account/account.py

```diff
@@ -269,15 +269,28 @@ def search(self, cr, uid, args, offset=0, limit=None, order=None,
                 order, context=context, count=count)
 
     def _get_children_and_consol(self, cr, uid, ids, context=None):
-        #this function search for all the children and all consolidated children (recursively) of the given account ids
-        ids2 = self.search(cr, uid, [('parent_id', 'child_of', ids)], context=context)
-        ids3 = []
-        for rec in self.browse(cr, uid, ids2, context=context):
-            for child in rec.child_consol_ids:
-                ids3.append(child.id)
-        if ids3:
-            ids3 = self._get_children_and_consol(cr, uid, ids3, context)
-        return ids2 + ids3
+        res = self._query_children_and_consol(cr, uid, ids, context=context)
+        if ids and not res:
+            self._parent_store_compute(cr)
```

Comment:
I have to agree about fixing symptoms :8ball: 

About accounts with no children triggering this, normally _get_children_and_consol returns the id of the parent, so when it doesn't return itself.

I've removed the _parent_store_compute. Besides, the CTE query goes around the parent_foo problem, thankfully.